src/puppet-web/firer.ts

```diff
@@ -59,9 +59,10 @@ const regexConfig = {
   ],
 
   roomJoinQrcode: [
-    /^"(.+)" joined the group chat via the QR Code shared by "?(.+?)".$/,
-    /^"(.+)" joined the group chat via "?(.+?)"? shared QR Code.$/,
```

Comment:
Why you get rid of `"?(.+?)"?`, and use `undefined` in your code, please explain.

I don't think this change is needed, and also I believe it's not a good choice to do this.